### PR TITLE
Off limb enhance again

### PR DESCRIPTION
**Pull request by wafels**
Changed to using pixel_to_world to avoid the deprecation warning.  Also changed to using LaTeX to show users you can use LaTeX.

### CODE REVIEW

File: examples/off_limb_enhance.py [hunk 1]
```diff
@@ -52,25 +52,32 @@
...
+
+plt.rc('text', usetex=True)
 plt.plot(rsun_array, y, label='data')
-label = r'fit=Aexp({:.2f}r)'.format(params[0])
+label = 'fit=$A\exp(${:.2f}$r)$'.format(params[0])
 plt.plot(rsun_array, np.exp(np.poly1d(params)(rsun_array)), label=label)
 plt.yscale('log')
 plt.ylabel('mean DN')
-plt.xlabel('radius r (Rsun)')
+plt.xlabel('radius r ($R_{\odot}$)')
```
Comment (Cadair) [inline examples/off_limb_enhance.py]: when using LaTeX specifically `\` it's best to use `r""` raw strings.

Comment (nabobalis) [reply examples/off_limb_enhance.py]: Like this now?

File: examples/off_limb_enhance.py [hunk 2]
```diff
@@ -52,25 +52,37 @@
...
-label = r'fit=Aexp({:.2f}r)'.format(params[0])
-plt.plot(rsun_array, np.exp(np.poly1d(params)(rsun_array)), label=label)
-plt.yscale('log')
-plt.ylabel('mean DN')
-plt.xlabel('radius r (Rsun)')
-plt.legend()
+best_fit = np.exp(np.poly1d(params)(rsun_array))
+label = r'best fit: {:.2f}$e^{{{:.2f}r}}$'.format(best_fit[0], params[0])
+plt.plot(rsun_array, best_fit, label=label)
+plt.yscale(r'log', fontsize=fontsize)
```
Comment (Cadair) [inline examples/off_limb_enhance.py]: why not `plt.rc` these fontsizes as well?

Comment (wafels) [reply examples/off_limb_enhance.py]: I could have, but then it would have changed the fontsize in the map plots.  Maybe I should do that too.

**Comment by larrymanley** [comment]
off_limb_enhance

**Comment by wafels** [comment]
I think the scaling has worked here, but maybe the vmax value has to be
lowered in order to see the effect.

*scaled_map.plot_settings['norm'] = colors.Normalize(vmin=10, vmax=10000)*

**Comment by larrymanley** [comment]
@wafels This is with vmax=3000.  Also, is there something wrong with the `scale_factor[r < 1] = 1`?
off_limb_enhance_vmax3000

**Comment by larrymanley** [comment]
This is with matched plot_settings for the original map (`aia.plot_settings['norm'] = colors.Normalize(vmin=10, vmax=3000`).
off_limb_enhance_vmax3000match

**Comment by wafels** [comment]
Hmmm. What does the plot of the observed mean radial DN look like with the
best fit overplotted?

**Comment by larrymanley** [comment]
@wafels
oleplot

**Comment by nabobalis** [comment]
This is what it was on the previous PR
figure_1

**Comment by wafels** [comment]
Might need to fit a higher degree polynomial to the 2nd example?
Also, the median value of the intensity in each annular region off limb
might give a better estimate of the function required to enhance the
offlimb emission.

I am working on a version of this function for sunkit-image - see
https://github.com/wafels/sunkit-image/blob/master/sunkit_image/offlimb_enhance.py
for a work in progress.

**Comment by nabobalis** [comment]
Actually when I run this example as it stands, I see the original result. 
figure_1

Can someone else run this for me?

**Comment by wafels** [comment]
Larry is using different data.

**Comment by larrymanley** [comment]
@wafels @nabobalis Regarding the plots, sorry I forgot to mention I'm not using the sample data.

**Comment by nabobalis** [comment]
Ah sorry about that, I should have noticed!

Well we could add a disclaimer to the example which parts would need to be modified for different data?
The examples do not need to be 100% infallible.

**Comment by wafels** [comment]
Use mathtext instead of using the LaTeX option (mathtext ships with matplotlib).  Also, change the scaling on the enhanced image to use a square root instead of the normalization currently used.

**Comment by wafels** [comment]
Now I get this - you can see that the on disk scaling is the same pre- and post-off limb enhancement.
bbb
aaa

File: examples/off_limb_enhance.py [hunk 3]
```diff
@@ -52,27 +52,42 @@
 params = np.polyfit(rsun_array[rsun_array < 1.5],
                     np.log(y[rsun_array < 1.5]), 1)
 
+###############################################################################
+# Tell matplotlib to use LaTeX for all the text, make the fontsize bigger, and
+# then plot the data and the fit.
+
+fontsize = 14
+plt.rc('xtick', labelsize=fontsize)
```
Comment (Cadair) [inline examples/off_limb_enhance.py]: Is there a way of doing this that doesn't mess with the rc?

**Comment by wafels** [comment]
Yes - using the ax.xaxis yadda yadda yadda format.  I'll do that instead.

**Comment by nabobalis** [comment]
There is problem with this being merged now?

**Comment by wafels** [comment]
Nope.  It's passed, right?

**Comment by nabobalis** [comment]
It passed.

**Pull request merged**